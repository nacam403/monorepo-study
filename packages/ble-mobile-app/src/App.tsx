import React, { useState } from 'react'
import { StyleSheet, SafeAreaView, View, Text, Button } from 'react-native'

import {
  scan,
  connect,
  retrieveServices,
  createBondWithRetry,
  write,
  read,
  disconnect,
} from './BleManagerWrapper'
import {
  WEIGHT_SCALE_SERVICE_UUID,
  DATE_TIME_CHARACTERISTIC_UUID,
  createDataTimeCharacteristicValue,
} from './WeightScaleUtil'

const blankPeripheral = {
  id: '-',
  name: '-',
}

const App = () => {
  const [discoveredPeripheral, setDiscoveredPeripheral] = useState(
    blankPeripheral,
  )

  return (
    <SafeAreaView style={style.safeAreaView}>
      <Button
        onPress={async () => {
          const discoveredPeripheral = await scan()
          setDiscoveredPeripheral(discoveredPeripheral || blankPeripheral)

          if (discoveredPeripheral) {
            await connect(discoveredPeripheral)
            await retrieveServices(discoveredPeripheral)

            // TODO if Android
            await createBondWithRetry(discoveredPeripheral)

            const currentDateTime = createDataTimeCharacteristicValue()
            await write({
              peripheral: discoveredPeripheral,
              serviceUUID: WEIGHT_SCALE_SERVICE_UUID,
              characteristicUUID: DATE_TIME_CHARACTERISTIC_UUID,
              bytes: currentDateTime,
            })

            const dateTimeArray = await read({
              peripheral: discoveredPeripheral,
              serviceUUID: WEIGHT_SCALE_SERVICE_UUID,
              characteristicUUID: DATE_TIME_CHARACTERISTIC_UUID,
            })
            console.log(dateTimeArray)
            await disconnect(discoveredPeripheral)
          }
        }}
        title="Paring"
      />

      <View>
        <Text>{`Discovered Peripheral ID: ${discoveredPeripheral.id}`}</Text>
        <Text>{`Discovered Peripheral Name: ${
          discoveredPeripheral.name
        }`}</Text>
      </View>
    </SafeAreaView>
  )
}

const style = StyleSheet.create({
  safeAreaView: {
    flex: 1,
    justifyContent: 'space-around',
  },
})

export default App
