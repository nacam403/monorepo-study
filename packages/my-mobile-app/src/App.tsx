import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { createStackNavigator, createAppContainer } from 'react-navigation'

const App = () => <AppContainer />

export default App

const Home = () => (
  <View style={styles.container}>
    <Text style={styles.text}>テキスト</Text>
    <Text style={styles.text}>テキスト</Text>
  </View>
)

const AppNavigator = createStackNavigator(
  {
    Home: {
      screen: Home,
    },
  },
  {
    defaultNavigationOptions: {
      title: 'ヘッダー',
    },
  },
)

const AppContainer = createAppContainer(AppNavigator)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  text: {
    fontSize: 25,
  },
})
