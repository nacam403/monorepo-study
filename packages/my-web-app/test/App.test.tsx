import React from 'react'
import renderer from 'react-test-renderer'
import App from '../src/App'

describe('App', () => {
  it('renders App component', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
