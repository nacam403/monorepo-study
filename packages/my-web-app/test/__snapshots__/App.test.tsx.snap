// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
<div>
  <h1>
    Hello, World!
  </h1>
</div>
`;
